examples/gazebo_tank/gazebo_tank.cc

```diff
@@ -31,12 +31,21 @@ bob_main(int argc, char **argv)
     gazebo::transport::NodePtr node = getGazeboNode();
 
     /************************************Gazebo setup end************/
-    //Initialize Gazebo camera display if -d arguement supplied
-    GazeboCameraInput cam(node, "/gazebo/default/differential_drive_robot/camera/link/camera/image");
-    Display display(cam);
-    if(argc >= 2 && strcmp(argv[1], "-d") == 0) {
+    std::unique_ptr<GazeboCameraInput> cam;
+    std::unique_ptr<Display> display;
```

Comment:
Might be worth swapping these two declarations as discussed in the email

Reply:
Swapped them but still getting the error 
`*** Error in `./gazebo_tank': double free or corruption (out): 0x000000000132dc30 ***`
when pressing the `B` button on the joystick.

Reply:
Hmm, well I don't have any ideas short term, so let's just merge it for now and worry about it later.

Good work!